java/src/main/java/com/logicalclocks/hsfs/FeatureView.java

```diff
@@ -249,67 +261,149 @@ public void deleteTag(String name) throws FeatureStoreException, IOException {
+  ) throws IOException, FeatureStoreException, ParseException {
+    TrainingDataset trainingDataset =
+        this.featureStore
+            .createTrainingDataset()
+            .name("") // name is set in the backend
+            .trainingDatasetType(TrainingDatasetType.IN_MEMORY_TRAINING_DATASET)
+            .eventStartTime(startTime)
+            .eventEndTime(endTime)
+            .splits(splits != null ?
+                splits.entrySet().stream().map(entry -> new Split(entry.getKey(), entry.getValue())).collect(
+                Collectors.toList()) : null)
+            .trainSplit(trainSplit)
+            .build();
     return featureViewEngine.getTrainingDataset(this, trainingDataset, Maps.newHashMap());
   }
 
   public TrainingDatasetBundle getTrainingDataset(
       Integer version, String startTime, String endTime, String description, Map<String, Float> splits,
       String trainSplit, StatisticsConfig statisticsConfig, Map<String, String> readOptions
-  ) {
-    TrainingDataset trainingDataset = new FeatureStore().createTrainingDataset().build();
+  ) throws IOException, FeatureStoreException, ParseException {
+    TrainingDataset trainingDataset =
+        this.featureStore
+            .createTrainingDataset()
+            .name("") // name is set in the backend
+            .trainingDatasetType(TrainingDatasetType.IN_MEMORY_TRAINING_DATASET)
+            .version(version)
+            .eventStartTime(startTime)
+            .eventEndTime(endTime)
+            .description(description)
+            .splits(splits != null ?
+                splits.entrySet().stream().map(entry -> new Split(entry.getKey(), entry.getValue())).collect(
+                    Collectors.toList()) : null)
+            .trainSplit(trainSplit)
+            .statisticsConfig(statisticsConfig)
+            .build();
     return featureViewEngine.getTrainingDataset(this, trainingDataset, readOptions);
   }
 
-  public void createTrainingDataset(
+  public TrainingDatasetBundle createTrainingDataset(
       String startTime, String endTime, DataFormat dataFormat, StorageConnector storageConnector
-  ) {
-    TrainingDataset trainingDataset = new FeatureStore().createTrainingDataset().build();
-    featureViewEngine.createTrainingDataset(this, trainingDataset, Maps.newHashMap());
+  ) throws IOException, FeatureStoreException, ParseException {
+    TrainingDataset trainingDataset =
+        this.featureStore
+            .createTrainingDataset()
+            .name("") // name is set in the backend
+            .eventStartTime(startTime)
+            .eventEndTime(endTime)
+            .dataFormat(dataFormat)
+            .storageConnector(storageConnector)
+            .build();
+    return featureViewEngine.createTrainingDataset(this, trainingDataset, Maps.newHashMap());
   }
 
   public void createTrainingDataset(
       String startTime, String endTime, DataFormat dataFormat, StorageConnector storageConnector,
```

Comment:
When calling this function fvUpdate.createTrainingDataset(null, null, CSV, null, splits, "train") it returns NULL instead of a TrainingDatasetBundle. I'm not sure why that is. But the training data gets successfully written to the HopsFS storage. Can you have a look at this?

Something a little unrelated, but I wonder why the training data is not coalesced before it's written. The resulting dataset is split into literally one CSV file per row.

Reply:
Good catch! `createTrainingDataset` should return `TrainingDatasetBundle` instead of `void`.
T
`coalesce` is optional and default to be false.